### ISSUE TITLE
qa: Decommissioning nodes

### ISSUE THREAD

**Issue by vivekmenezes** [cockroachdb/cockroach#21882]

**Comment by tschottdorf** [comment]
@m-schneider one thing I'd like you to check is that when you decommission a node (whether it is down before you do that or not), the rest of the cluster stops connecting to it. In particular, it shouldn't show up in any of the debug pages, ui, ..., and there shouldn't be log entries for its IP.

**Comment by tschottdorf** [comment]
In a four node cluster, decommissioning the first node in absentia, the graphs show one replica remains on that node forever (this isn't actually true; `select replicas from crdb_internal.ranges;` shows all replicas on the remaining three nodes):

<img width="869" alt="image" src="https://user-images.githubusercontent.com/5076964/37076227-ae3fabe4-21a3-11e8-927c-5db4a143ad05.png">

@couchand is that a ui bug? The node is down, so it can't possibly write that time series.

**Comment by tschottdorf** [comment]
On the plus side, after waiting for the store dead timeout, the node disappeared from the graphs and wasn't shown as dead any more, which was expected and worked fine! 👍

**Comment by tschottdorf** [comment]
Some graphs from my WIP node decommissioning test (gracefully quit first of three nodes, decommission it, wipe it, bring it back, wait a minute). The test runs a max-qps=500 kv on the side. This is all on my laptop.

TLDR is that it looks pretty good. We seem to be doing a decent enough job a) quitting gracefully and b) not messing things up during the rebalances when the node comes back:

<img width="963" alt="image" src="https://user-images.githubusercontent.com/5076964/37085853-1d8ea14c-21c4-11e8-946c-527d67c8adf6.png">

<img width="882" alt="image" src="https://user-images.githubusercontent.com/5076964/37085913-47c0e9e8-21c4-11e8-9cdd-5ece022a4b08.png">

There are occasionally dips like these:

<img width="876" alt="image" src="https://user-images.githubusercontent.com/5076964/37085948-6583fb8c-21c4-11e8-9897-2347a861f72f.png">

correlated with a spike in node heartbeat latency (~500ms)

That's something to investigate. Potentially this is an artifact of running on my laptop while I do other stuff, though.

**Comment by couchand** [comment]
> show one replica remains on that node forever

This is a known charting issue (#23480 I think, though it's manifesting a bit strangely in your case...) If you look at the big dot for node 1, it's not at the same point in time as the other ones.  I think all that's happening is that the last recorded data point for that series was 1.  It might be a good usability win to make sure decommissioned nodes always record one more datapoint before going offline, so that they are forever recorded as being clean of all replicas.  I've recorded #23525 to keep track of that idea.

**Comment by tschottdorf** [comment]
Thanks @couchand! I've commented on https://github.com/cockroachdb/cockroach/issues/23525 directly.
